### ISSUE TITLE
Sporadic missing keystrokes

### ISSUE THREAD

**Issue by insunaa** [Frogging-Family/wine-tkg-git#498]
Using wine-tkg 3c54897511bdeb25c1a8ac85e926614e2e3d1091 I noticed that on occasion some keypresses are just not registered by the application I'm using. In my particular use case that's in the game world of warcraft. I can't reliably reproduce it, it seem to happen after a combination of mouse inputs and keyboard inputs.

I notice this because sometimes I hit movement/direction keys and the character just does not start moving in this direction.
I don't know in which version this went wrong, I only know it's definitely earlier than 6.16, earlier than 6.10 too I believe.
The last Version that I *know* works is wine-tkg 6.0a

**Comment by shaumux** [comment]
Have you checked the proton logs and the kernel logs for any clues?
Posting them here would be helpful in debugging if there any issues with proton

**Comment by insunaa** [comment]
@shaumux Thre are no proton logs because it's just wine-tkg not proton-tkg
I'll try checking the kernel logs, but I'm not sure I'll see much there, but worth a try! :)

**Comment by insunaa** [comment]
OK, at least for the DX12 issue I have a hint, and that is with wine-tkg 6.0a I can achieve the same non-functioning state by adding the environment variable `VKD3D_CONFIG=dxr` this breaks the dx12 compatibility in the exact same way.

Is dxr somehow an environment variable that wine-tkg 6.16 automatically passes to vkd3d?

**Comment by Tk-Glitch** [comment]
No, dxr isn't enabled by default in wine-tkg (nor is vkd3d). I have no problem running wow and other d3d12 games on 6.16 myself (with vkd3d-proton and without lutris) Make sure you're building wine with `_use_vkd3dlib="false"` set in your config, and ideally no vkd3d package installed.

**Comment by Tk-Glitch** [comment]
Also see https://github.com/HansKristian-Work/vkd3d-proton/issues/764

**Comment by insunaa** [comment]
That was indeed the issue for the dx12 thing, thank you and sorry for taking your time.
Any idea what I could use to log the missing keystroke events? `xev` detects all keystrokes correctly.

I don't have exclusive mouse capturing enabled, in 6.0a this doesn't cause a problem, but could it be that between then and 6.16 something changed that would require exclusive mouse capturing? (I have a dual-monitor setup)

**Comment by Tk-Glitch** [comment]
> thank you and sorry for taking your time

You're welcome. The more I'm aware of issues and solutions, the faster I can point them out when needed, so definitely not a time loss imho ;)

There has been quite a few changes to input handling in 8+ months :stuck_out_tongue: 
High polling rate devices can lead to such results when tickrate isn't high enough on the kernel side. Reducing the device polling rate or rebuilding your kernel with a higher tickrate might help.

This might be of interest to you: https://forums.lutris.net/t/wow-mouse-btn-on-hold-disturbs-keyboard-inputs/8172

**Comment by insunaa** [comment]
That is an issue I had before too and that particular issue was also fixed with the provided commands.

I'm currently using 5.13.13-195-tkg-pds with a 100hz tickrate, if I may ask what tickrate would you recommend?

**Comment by Tk-Glitch** [comment]
100 Hz is extremely low for interactive usage like gaming. I'd recommend trying 500+, 1000 Hz offering the lowest input latency which is likely to help here.

**Comment by insunaa** [comment]
OK, 1000hz kernel does not solve the issue, but it *feels* like it occurs less often. No empirical data on that, tho :(

**Comment by Tk-Glitch** [comment]
What kind of mouse/keyboard combo are you using?

**Comment by insunaa** [comment]
I'm using a gmmk as a keyboard and a razer naga (the one with the side keypad) as a mouse.
There have been issues with this specific combo with Gnome's mutter compositor, but there's a discarded pull-request that I cherry-picked into my mutter compilation that fixes that (switching between keyboards causes a system hang for ~20ms)

Both are USB 2.0, the gmmk uses n-key rollover

**Comment by insunaa** [comment]
Is there anything useful I could do to help you or anybody identify the issue?
Any tools I could use, any specific logs I could check?
journald and dmesg haven't produced anything meaningful so far... I'm getting a Mode SixtyFive keyboard in ~November, if you suspect it to be a keyboard issue I can test it then. I don't have a second mouse, but I could try reproducing the issue without a mouse plugged in? Tho so far I've been pretty sure that mouse inputs (such as move mouse-right-click-drag for viewport moving) were involved in making this issue occur.

I think the KBD67 Lite keyboard I built for someone has a 100hz polling rate, I'll test it with that for a while, then.

**Comment by Tk-Glitch** [comment]
I guess the next step will be to open a ticket on winehq. Maybe @rbernon knowns what's going on.

**Comment by rbernon** [comment]
Not really, but anything is possible and for instance that quickly pressed key cancel each other. It's hard to tell like that.
